File: src/main/java/gregtechlite/gtlitecore/mixins/EarlyMixinLoader.java
package gregtechlite.gtlitecore.mixins;

import gregtechlite.gtlitecore.api.GTLiteValues;
import net.minecraftforge.fml.common.Loader;
import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin;
import org.jetbrains.annotations.Nullable;
import zone.rong.mixinbooter.IEarlyMixinLoader;

import java.util.List;
import java.util.Map;

@IFMLLoadingPlugin.Name(GTLiteValues.MOD_NAME)
@IFMLLoadingPlugin.MCVersion(Loader.MC_VERSION)
public final class EarlyMixinLoader implements CustomMixinLoader, IFMLLoadingPlugin, IEarlyMixinLoader
{
    @Override
    public List<String> getMixinConfigs()
    {
        return createMixinConfigs("minecraft");
    }

    @Override
    public String @Nullable[] getASMTransformerClass()
    {
        return null;
    }

    @Override
    public @Nullable String getModContainerClass()
    {
        return null;
    }

    @Override
    public @Nullable String getSetupClass()
    {
        return null;
    }

    @Override
    public void injectData(Map<String, Object> data) {}

    @Override
    public @Nullable String getAccessTransformerClass()
    {
        return null;
    }
}

File: src/main/kotlin/gregtechlite/gtlitecore/mixins/EarlyMixinLoader.kt
package gregtechlite.gtlitecore.mixins

import gregtechlite.gtlitecore.api.MOD_NAME
import net.minecraftforge.fml.common.Loader
import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin
import zone.rong.mixinbooter.IEarlyMixinLoader

@IFMLLoadingPlugin.Name(MOD_NAME)
@IFMLLoadingPlugin.MCVersion(Loader.MC_VERSION)
class EarlyMixinLoader : CustomMixinLoader, IFMLLoadingPlugin, IEarlyMixinLoader
{
    override fun getMixinConfigs(): List<String> = createMixinConfigs("minecraft")

    override fun getASMTransformerClass(): Array<out String>? = null

    override fun getModContainerClass(): String? = null

    override fun getSetupClass(): String? = null

    override fun injectData(data: Map<String, Any>) {}

    override fun getAccessTransformerClass(): String? = null
}